enum_tools/utils.py

```diff
@@ -235,6 +251,13 @@ def fast_dns_lookup(names, nameserver, nameserverfile, callback='', threads=5):
 
         # Update the status message
         sys.stdout.flush()
+        if LOGFILE_FMT == 'json':
+            data = {'type':'progress',
+                    'source':'fast_dns_lookup',
+                    'data': {"total": total, "current": current}}
+            if callback:
+                data['source'] = callback.__name__.replace("print_", "")
```

Comment:
Can you help me understand what is happening here? What gets logged if JSON is being used, relative to the printed progress bar?

Reply:
As mentioned in the other comment, each JSON line is on its own - without any relation to the previous one

It should be also considered its own "dataset" - some are progress, some are outcome - its a bit like how old-school print line status was done but with a JSON output

To make the whole output more computer friendly, a lot of code changes are necessary